azure/scope/machinepool.go

```diff
@@ -222,10 +222,8 @@ func (m *MachinePoolScope) ScaleSetSpec(ctx context.Context) azure.ResourceSpecG
 	}
 
 	if m.cache != nil {
-		if m.HasReplicasExternallyManaged(ctx) {
-			spec.ShouldPatchCustomData = m.cache.HasBootstrapDataChanges
```

Comment:
For my own understanding: why are we removing the check for `m.HasReplicasExternallyManaged(ctx)`?

Reply:
further below in `ScaleSetSpec#existingParameters` we determine whether we need to set `VirtualMachineProfile` to `nil` based on whether there are model changes or the `ShouldPatchCustomData` flag is being set. This flag will now be used for all kind of autoscaling, not just externally managed. We don't know otherwise if the custom data changed or not (since, when fetching a VMSS, you don't receive the customData for security reasons).